src/ray/core_worker/core_worker.cc

```diff
@@ -832,6 +832,8 @@ void CoreWorker::Exit(
   RAY_CHECK_OK(
       local_raylet_client_->NotifyDirectCallTaskBlocked(/*release_resources*/ true));
 
+  RAY_LOG(DEBUG) << "Exit signal received, remove all local references.";
+  reference_counter_->ReleaseAllLocalReferences();
```

Comment:
Could you add a comment here to explain the reason if this step? And please add it to the PR description as well.